integration_tests/tests/transactionCSV-to-s3-event-tests.ts

```diff
@@ -1,97 +1,70 @@
+import { mockIdpCsvData } from "../../src/handlers/int-test-support/helpers/mock-data/csv";
 import {
   deleteS3Objects,
   getS3Object,
   listS3Objects,
   putS3Object,
-  S3Object,
 } from "../../src/handlers/int-test-support/helpers/s3Helper";
 
-const prefix = resourcePrefix();
-const folderPrefix = "btm_transactions";
-const bucketName = `${prefix}-storage`;
+const transactionsDirectory = "btm_transactions";
+const outputBucket = `${resourcePrefix()}-storage`;
+const inputBucket = `${resourcePrefix()}-transaction-csv`;
 
-const checkS3BucketForEventIds = async (): Promise<boolean> => {
-  const result = await listS3Objects({
-    bucketName,
-    prefix: folderPrefix,
-  });
-  if (
-    result.Contents === undefined ||
-    result.Contents.length !==
-      testCases.filter((data) => data.expectedPath === "happy").length
-  ) {
-    return false;
-  } else {
-    return true;
-  }
-};
+const { testCases, csv } = mockIdpCsvData();
 
-describe("\n Given a csv with event data is uploaded to the transaction csv bucket", () => {
+describe("Given a csv with event data is uploaded to the transaction csv bucket", () => {
   beforeAll(async () => {
     // First delete all transactions in btm_transaction
-    await deleteS3Objects({ bucketName, prefix: folderPrefix });
-    // Add data to build the test csv
-    const augmentedData = addFauxDataToTestCases(testCases, {
-      Timestamp: "2023-01-01T00:27:41.186Z",
-      "RP Entity Id": "fake rp entity id",
+    await deleteS3Objects({
+      bucketName: outputBucket,
+      prefix: transactionsDirectory,
     });
-    // Filter and rename data in the test csv
-    const csvString = objectsToCSV(augmentedData, {
-      filterKeys: ["expectedPath"],
-      renameKeys: new Map([
-        ["givenEntity", "Idp Entity Id"],
-        ["givenLoa", "Minimum Level Of Assurance"],
-        ["givenStatus", "Billable Status"],
-        ["expectedEventId", "Request Id"],
-      ]),
-    });
-    // Upload the csv file to S3 transaction csv bucket
-    const testObject: S3Object = {
-      bucket: `${prefix}-transaction-csv`,
-      key: `fakeBillingReport.csv`,
-    };
-    await putS3Object({ target: testObject, data: Buffer.from(csvString) });
-  });
+    // Add data to build the test csv
 
-  it("stores valid events in the storage/btm_transactions/yyyy-mm-dd folder", async () => {
-    const checkEventsExistsInS3 = await waitForTrue(
-      checkS3BucketForEventIds,
-      1000,
-      10000
+    // Upload the csv file to S3 transaction csv bucket
+    await putS3Object({
+      target: {
+        bucket: inputBucket,
+        key: `fakeBillingReport.csv`,
+      },
+      data: csv,
+    });
+    await poll<ListObjectsCommandOutput>(
+      async () =>
+        await listS3Objects({
+          bucketName: outputBucket,
+          prefix: transactionsDirectory,
+        }),
+      (result) =>
+        result.Contents?.length !== undefined && result.Contents?.length > 0
```

Comment:
Here you can see the two elements of the old `checkS3BucketForEventIds` function. `listS3Objects` is the thing we want to keep doing over and over until we get the desired outcome. We'll know we get our desired outcome when there are items in the `result.Contents` array.